packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java

```diff
@@ -82,43 +82,31 @@ private File resizedImage(
   }
 
   private SizeFCompat calculateTargetSize(
-      @NonNull Double originalWidth,
-      @NonNull Double originalHeight,
+      @NonNull double originalWidth,
+      @NonNull double originalHeight,
       @Nullable Double maxWidth,
       @Nullable Double maxHeight) {
 
+    double aspectRatio = originalWidth / originalHeight;
+
     boolean hasMaxWidth = maxWidth != null;
     boolean hasMaxHeight = maxHeight != null;
 
-    Double width = hasMaxWidth ? Math.min(originalWidth, maxWidth) : originalWidth;
-    Double height = hasMaxHeight ? Math.min(originalHeight, maxHeight) : originalHeight;
+    Double width = hasMaxWidth ? Math.min(originalWidth, Math.round(maxWidth)) : originalWidth;
+    Double height = hasMaxHeight ? Math.min(originalHeight, Math.round(maxHeight)) : originalHeight;
 
     boolean shouldDownscaleWidth = hasMaxWidth && maxWidth < originalWidth;
     boolean shouldDownscaleHeight = hasMaxHeight && maxHeight < originalHeight;
     boolean shouldDownscale = shouldDownscaleWidth || shouldDownscaleHeight;
 
     if (shouldDownscale) {
-      double downscaledWidth = (height / originalHeight) * originalWidth;
-      double downscaledHeight = (width / originalWidth) * originalHeight;
-
-      if (width < height) {
-        if (!hasMaxWidth) {
-          width = downscaledWidth;
-        } else {
-          height = downscaledHeight;
-        }
-      } else if (height < width) {
-        if (!hasMaxHeight) {
-          height = downscaledHeight;
-        } else {
-          width = downscaledWidth;
-        }
+      double downScaledWidth = height * aspectRatio;
+      double downScaledHeight = width / aspectRatio;
```

Comment:
I found these names pretty confusing. What about `widthForMaxHeight` and `heightForMaxWidth`?

Reply:
Not sure yours are any less confusing, but they are more descriptive.